app/helpers/events_helper.rb

```diff
@@ -1,9 +1,10 @@
 module EventsHelper
   def set_event
+    not_found && return if (params[:year].to_i >= current_year)
```

Comment:
I think need to add rescue here, what if the user input a non number value for year?

Reply:
Thanks for reminding, I actually did not notice as well, but magically it works fine (render not_found page) XD if you input non-integer year. After further investigation, https://apidock.com/ruby/String/to_i, you can check the details here about how to_i method works.